tests/test_markdown.py

```diff
@@ -0,0 +1,86 @@
+import io
+import os
+import textwrap
+
+from readme_renderer.markdown import render
+
+
+def test_simple():
+    markdown_markup = 'Hello'
+    out = render(markdown_markup)
+    assert out == '<p>Hello</p>'
+
+
+def test_url_no_link():
+    markdown_markup = 'http://mymalicioussite.com/'
+    out = render(markdown_markup)
+    expected_html = '<p>http://mymalicioussite.com/</p>'
+    assert out == expected_html
+
+
+def test_iframe():
+    markdown_markup = """\
+        <iframe src="http://mymalicioussite.com/">Click here</iframe>
+    """.strip()
+    out = render(markdown_markup)
+    expected_html = ''.join([
+        '&lt;iframe src="http://mymalicioussite.com/"&gt;'
+        'Click here&lt;/iframe&gt;'])
+    assert out == expected_html
+
+
+def test_script():
+    markdown_markup = textwrap.dedent("""\
+        <script>
+            alert("Hello");
+        </script>""")
+    out = render(markdown_markup)
+    expected_html = textwrap.dedent("""\
+        &lt;script&gt;
+            alert("Hello");
+        &lt;/script&gt;""")
+    assert out == expected_html
+
+
+def test_a_tag_gets_nofollow():
+    markdown_markup = '<a href="http://mymalicioussite.com/">Click here</a>'
+    out = render(markdown_markup)
+    expected_htmls = [
+        ''.join(['<p><a rel="nofollow" href="http://mymalicioussite.com/">',
+                 'Click here</a></p>']),
+        ''.join(['<p><a href="http://mymalicioussite.com/" rel="nofollow">',
+                 'Click here</a></p>']),
+    ]
+    assert out in expected_htmls
+
+
+def test_smart_strong():
+    markdown_markup = 'Text with double__underscore__words.'
+    out = render(markdown_markup)
+    expected_html = '<p>Text with double__underscore__words.</p>'
+    assert out == expected_html
+
+
+def test_github_backtick_syntax_for_code():
+    _do_test_with_files('code')
+
+
+def test_headings_and_paragraphs():
+    _do_test_with_files('headings_and_paragraphs')
+
+
+def test_misc():
+    _do_test_with_files('misc')
+
+
+def _do_test_with_files(test_name):
+    md_markup = read('{0}.md'.format(test_name))
+    expected_html = read('{0}.html'.format(test_name))
+    out = render(md_markup)
+    assert out == expected_html.rstrip()
+
+
+def read(fn):
+    path = os.path.join(os.path.dirname(__file__), 'fixtures', 'markdown', fn)
+    with io.open(path, encoding='utf-8') as f:
```

Comment:
I'm not sure backporting the `encoding` parameter to Python 2's `open` is worth the effort since Python 2 is end of life. Let's just stick, [like](https://github.com/search?l=Python&q=%22import+io%22&type=Code&utf8=%E2%9C%93) [everyone](http://stackoverflow.com/questions/10971033/backporting-python-3-openencoding-utf-8-to-python-2) [else](http://python-future.org/compatible_idioms.html#file-io-with-open), to a **simple** and **temporary** "import io".

Also, it will be easier to port Python 2 using `io.open` than Python 2 using `open`, as `Python 3`'s open [is](https://docs.python.org/3/library/io.html#io.open) `io.open`.